src/services/maci/constants.ts

```diff
@@ -1,39 +1,40 @@
 import {
-    EContracts,
-    PollFactory__factory as PollFactoryFactory,
-    MessageProcessorFactory__factory as MessageProcessorFactoryFactory,
-    TallyFactory__factory as TallyFactoryFactory,
-    MACI__factory as MACIFactory,
-    ConstantInitialVoiceCreditProxy__factory as ConstantInitialVoiceCreditProxyFactory,
-    EASGatekeeper__factory as EASGatekeeperFactory,
-    Verifier__factory as VerifierFactory,
-    PoseidonT3__factory as PoseidonT3Factory,
-    PoseidonT4__factory as PoseidonT4Factory,
-    PoseidonT5__factory as PoseidonT5Factory,
-    PoseidonT6__factory as PoseidonT6Factory,
-    VkRegistry__factory as VkRegistryFactory,
-  } from "maci-cli/sdk";
+  PollFactory__factory as PollFactoryFactory,
+  MessageProcessorFactory__factory as MessageProcessorFactoryFactory,
+  TallyFactory__factory as TallyFactoryFactory,
+  MACI__factory as MACIFactory,
+  ConstantInitialVoiceCreditProxy__factory as ConstantInitialVoiceCreditProxyFactory,
+  EASGatekeeper__factory as EASGatekeeperFactory,
+  Verifier__factory as VerifierFactory,
+  VkRegistry__factory as VkRegistryFactory,
+} from "maci-cli/sdk";
 import type { IAbi } from "./types";
 
 /**
  * Constant variables
  */
 export const STATE_TREE_SUB_DEPTH = 2;
+export const STATE_TREE_DEPTH = 10;
+export const INT_STATE_TREE_DEPTH = 1;
+export const MESSAGE_TREE_DEPTH = 2;
+export const MESSAGE_BATCH_DEPTH = 1;
+export const VOTE_OPTION_TREE_DEPTH = 2;
+
+export const POSEIDON_T3_ADDRESS = "0x8d88DD98AD662Ff57d3971a686f5971fd3b01a59";
```

Comment:
hardcoding these here will also make it work only for op sepolia

Reply:
also I keep getting PoseidonHashLibrariesNotLinked when deploying the maci contract

Reply:
So actually there are several things should be input by the user:
1. NEXT_PUBLIC_ROUND_ID
2. NEXT_PUBLIC_CHAIN_NAME (could be a dropdown menu)
3. NEXT_PUBLIC_EASSCAN_URL (could be a mapping, the key is the CHAIN_NAME, and the url is the value)
4. NEXT_PUBLIC_ROUND_ID (actually this refer to "voting round name", we should store this value mapped with the pollId)
5. NEXT_PUBLIC_APPROVAL_SCHEMA
6. NEXT_PUBLIC_EAS_CONTRACT_ADDRESS & NEXT_PUBLIC_EAS_SCHEMA_REGISTRY_ADDRESS (also a mapping regarding to the CHAIN_NAME)
7. POSEIDON_T3/4/5/6_ADDRESS (also a mapping regarding to the CHAIN_NAME? or could be customized by getting input from the user)
8. STATE_TREEE_DEPTH
9. POLL_DURATION (could be choosing the end date, just like what maci-starter-kit does)

since there are so many things to do, that's why I'm thinking about putting this in a separate PR, and keeping this PR purely making sure the functions in the `MaciService.deploy...` functions could work. How do you think?

Reply:
that could work, but then what if we don't merge this to main yet and work on a separate branch?